### PR TITLE
added qrcode scanning capability

### PR DESCRIPTION
**Pull request by nathfavour**
The module is now able to scan qrcodes. Each of these commands accept either path to qrcode file(--use-qr) or path to qrcode link file (--use-qrl)

totp-cli add-token --use-qr (requires user input for namespace and password)

totp-cli add-token --use-qr --silq (doesn't require user input for namespace, account and token. just password)

top-cli add-token --use-qrl (link to qrcode link file e.g otpauth://totp/account?secret=asdfasdfasdfasdf&issuer=namespace)

### CODE REVIEW

**Comment by nathfavour** [comment]
you will also notice that there is an option for batch qrcode scanning (by passing directory paths), but that proved a little difficult because of the modularity of the library, and for now, even if a directory path is passed, only the first qrcode file is actually processed.

There is a possibility of batch processing functionality in the future, tho

File: internal/cmd/add_token.go [hunk 1]
```diff
@@ -1,23 +1,176 @@
...
 
-	s "github.com/yitsushi/totp-cli/internal/storage"
-	"github.com/yitsushi/totp-cli/internal/terminal"
-)
+func installZbarimg() error {
+    var cmd *exec.Cmd
+
+    switch os := runtime.GOOS; os {
+    case "linux":
+        cmd = exec.Command("sudo", "apt-get", "install", "zbar-tools")
```
Comment (yitsushi) [inline internal/cmd/add_token.go:42]: That's a huge no. Just as a starter, I use Arch on one machine, and Manjaro on another one.

My view on thins, a standalone software should NEVER install anything unless that's the purpose of the job.

**Review by yitsushi — COMMENTED** [review]
First of all, thank you for the contribution, but I have to reject this idea.

That would unnecessarily complicate the tool adding dependencies they should not need to have. This tool is used from CLI, in a lot of cases without any graphical environment on the machine. All providers offer an option to copy the code instead of reading a QR code, at least I never met one who didn't.

As I wrote under an issue about syncing. This tool is intended to be a simple tool, with as few as possible dependencies. The tool could have encrypted notes, encrypted data with custom fields. No, that would defeat the purpose of a small simple cli tool that follows (as much as it can) the unix philosophy.

Installing extra dependencies after the software was installed is another issue.
 - First of all, it shouldn't do it.
 - Second it doesn't handle (and I think it's nearly impossible) all possible OS/Distro options.
 - 3rd, the installed dependency is not managed by anything, if someone removes this tool, nothing will clean up the installed dependency for them. If something is a dependency if should be marked as dependency, either in go, or in the install docs and notify all package managers (for example [macports](https://ports.macports.org/port/totp-cli/details/), Homebrew, or AUR). As the application's main focus and sole purpose is to be a CLI only, but fully functional application, adding a graphical dependency is not something I would like to see.
 - And 4tf, shelling out to call an external binary shouldn't be necessary and it's kind of ugly. Go has capabilities to read images from screen or camera. If we ever want to add this functionality, I would suggest to go on that route.
 
About the sub-shell. the whole `zbarimg` can be simply applied now with pipes. It will not work with namespace (still can with a wrapper shell script). A few points that can help to write your wrapper script:
 - You can set `TOTP_PASS` to it does not ask. For example your script starts with `export TOTP_PASS=$(age --decrypt --identity ~/.age/key.txt ~/.config/totp-cli/totp-password.age)`
 - You can pipe the code to the application: `zbarimg ... | totp-cli add-token personal randomaccount`
 - You can parse the verbose output and populate the totp-cli call arguments: `echo "${TOKEN}" | totp-cli add-token "${NAMESPACE}" ${ACCOUNT}`
 - You can do all of that in a for loop.
 
That was you achieve the same results, but without bloating the application. That extra script can be part of the repo as tooling tho.

I hope my stance is understandable and not too upsetting.

File: internal/cmd/add_token.go (hunk 1 above)
Comment (nathfavour) [reply internal/cmd/add_token.go:42]: Thanks a LOT @yitsushi. Actually, I didn't really consider these factors, I was just concerned with it working as soon as I could lol 

I've learnt a lot about good software practices, thank you.

**Comment by nathfavour** [comment]
Ok, I'm kinda lazy lol, I will just continue using my current qrcode scanning method locally, then if someone doesn't apply the method you suggested, I will redo it as you suggested

**Comment by yitsushi** [comment]
Closing this PR based on the discussion / suggestions above.

**Pull request closed without merging**